src/timer.rs

```diff
@@ -14,18 +14,44 @@ mod inner {
 
     /// same as `after` but return () to make it compatible as previous
     pub fn sleep(duration: Duration) -> Sleeper {
-        Sleeper(after(duration))
+        Sleeper::after(duration)
+    }
+
+    pub enum SleeperFired {
+        Fired,
+        NotFired,
+        // used when timer is based on interval
+        // or anything that could be fired multiple times
+        None,
     }
 
     #[pin_project]
-    pub struct Sleeper(#[pin] Timer);
+    pub struct Sleeper {
+        #[pin]
+        timer: Timer,
```

Comment:
Timer, when using `after` now only fires once. This makes or CI test to fail. I implemented support to this `Sleeper` wrapper to keep previous behavior, but not sure if we need to do this or update the test to reflect the new behavior.

Not sure if in fluvio or in other projects we rely on `sleep` being able to fire multiple times

Reply:
Hmm.  sleeper should only fire once

Reply:
ok, I reverted the commit but also changed the test